Quick-Start [DA]/MVP/MVP+Makefile.md

```diff
@@ -1 +1,274 @@
+
+```bash
+$ touch Makefile
+```
+- Open the Makefile and add the following content:
+<!--Problem: https://dba.stackexchange.com/questions/328624/makefile-issue-in-creating-postgres-extension
+Had an issue with the make install command:
+```makefile
+DATA = my_extension--1.0.sql
+
+PG_CONFIG  ?= pg_config
+PGXS := $(shell $(PG_CONFIG) --pgxs)
+include $(PGXS)
+```
+Solution: Makefile is lacking the entry `EXTENSION = my_extension`
+Reason: That's why `make install` doesn't copy the `*.control` file, and that's why `CREATE EXTENSION` fails.
+
+NOTE: make: Nothing to be done for 'all' 
+Makefile doesn't have any MODULE or other binary to build. You can go directly to the make install step
+-->
+
+```makefile
+EXTENSION = my_extension
+DATA = my_extension--1.0.sql
+
+PG_CONFIG  ?= pg_config
+PGXS := $(shell $(PG_CONFIG) --pgxs)
+include $(PGXS)
+```
+
+### Step 5: Build and Install the Extension
+- To build and install the extension, run the following commands:
+
+```bash
+$ make
+$ make install
+```
+- The "make" command compiles the extension code using the Makefile
+- The "make install" command copies the necessary files to the appropriate PostgreSQL directories.
+
+### Step 6: Create the Extension in PostgreSQL
+- Connect to your PostgreSQL database and execute the following SQL statement to create the extension:
+
+```sql
+CREATE EXTENSION IF NOT EXITS "my_extension";
+```
+- The CREATE EXTENSION statement is executed in PostgreSQL to create the extension.
+
+## Implementation of the Extension:
+Once you have successfully implemented the `my_extension` in your PostgreSQL database, you can use it as follows:
+
+### 1. Query the `my_table` table:
+   - You can query the `my_table` table like any other table in your database. For example, you can retrieve all rows from the table using the following SQL query:
+   ```sql
+   SELECT * FROM my_table;
+   ```
+#### Output:
+```sql
+id | name
+----+------
+(0 rows)
+```
+Remember to replace `my_table` with the actual table name you have chosen, and adjust the function call as needed based on your specific implementation.
+
+## Regression Testing:
+
+### Step 1: Create the Extension SubDirectory for Regression SQL File:
+- Create a subdirectory for your regression file. Name it `sql`
+
+```bash
+$ mkdir sql
+$ cd sql
+```
+-   Create a new file called `my_extension_regression.sql`.
+
+```bash
+$ touch my_extension_regression.sql
+```
+
+- Open the `my_extension_regression.sql` file and add the following content keeping in mind these instructions:
+
+  -   Write SQL statements that test the functionality of your `my_extension`.
+
+  -   Include a combination of `CREATE`, `INSERT`, `UPDATE`, `DELETE`, and `SELECT` statements to cover various scenarios.
+
+  -   Make sure to test both positive and negative scenarios to validate the behaviour of your extension.
+
+```sql
+-- regression test script for my_extension
+-- Create the my_table table
+CREATE TABLE my_table (
```

Comment:
Why would you be creating this table when the extension already creates it? The first thing the script should do is `CREATE EXTENSION my_extension`.

Reply:
Fixed it